### PR TITLE
7478 normalize Selenium resource files creation

### PR DESCRIPTION
**Pull request by DanielWilczek**

### CODE REVIEW

File: G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs [hunk 1]
```diff
@@ -40,14 +35,15 @@ public override void LoadDlls()
...
+            var driversDictionary = new Dictionary<string, byte[]>()
             {
                 { "chromedriver.exe", Resources.chromedriver },
                 { "geckodriver.exe", Resources.geckodriver },
                 { "IEDriverServer.exe", Resources.IEDriverServer },
                 { "MicrosoftWebDriver.exe", Resources.MicrosoftWebDriver }
             };
-            foreach (var exe in exeList)
+            foreach (var exe in driversDictionary.Where(e => !File.Exists(Path.Combine(unpackfolder, e.Key))
+            || e.Value.Length != new FileInfo(Path.Combine(unpackfolder, e.Key)).OpenRead().Length))
```
Comment (lfronczyk) [inline G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: `FileInfo` should already contain size of the file, you don't need to open it.
DRY - `Path.Combine(unpackfolder, e.Key)`; move the lambda (or some parts of it) into a method.
`driversDictionary` with `exe` elements => `driversDictionary` => `drivers`, `exe` => `driver`

File: G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs [hunk 2]
```diff
@@ -56,7 +52,7 @@ private void UnpackDrivers()
                         stream.Write(exe.Value, 0, exe.Value.Length);
                     }
                 }
-                catch { }                
+                catch (Exception ex) { RobotMessageBox.Show(ex.Message); }
```
Comment (lfronczyk) [inline G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: 👍 
Consider rethrow of the exception.

Comment (DanielWilczek) [reply G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: Can't re-throw at this point, if we get exception at 1st file than other files are not going to be extracted at all.

File: G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs [hunk 3]
```diff
@@ -40,14 +35,14 @@ public override void LoadDlls()
...
-            Dictionary<string, byte[]> exeList = new Dictionary<string, byte[]>()
+            var driversDictionary = new Dictionary<string, byte[]>()
             {
                 { "chromedriver.exe", Resources.chromedriver },
                 { "geckodriver.exe", Resources.geckodriver },
                 { "IEDriverServer.exe", Resources.IEDriverServer },
                 { "MicrosoftWebDriver.exe", Resources.MicrosoftWebDriver }
             };
-            foreach (var exe in exeList)
+            foreach (var exe in driversDictionary.Where(e => !DoesFileExist(unpackfolder, e.Key) || DoesFileSameLenght(e.Value.Length, unpackfolder, e.Key)))
```
Comment (lfronczyk) [inline G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: AreFilesTheSameSize

File: G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs [hunk 4]
```diff
@@ -56,8 +51,18 @@ private void UnpackDrivers()
...
         }
+
+        private bool DoesFileExist(string folder, string fileName)
+        {
+            return File.Exists(Path.Combine(folder, fileName));
+        }
+
+        private bool DoesFileSameLenght(int lenght, string folder, string fileName)
+        {
+            return lenght != new FileInfo(Path.Combine(folder, fileName)).Length;
```
Comment (lfronczyk) [inline G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: length

File: G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs [hunk 5]
```diff
@@ -40,14 +35,14 @@ public override void LoadDlls()
...
-            Dictionary<string, byte[]> exeList = new Dictionary<string, byte[]>()
+            var driversDictionary = new Dictionary<string, byte[]>()
             {
                 { "chromedriver.exe", Resources.chromedriver },
                 { "geckodriver.exe", Resources.geckodriver },
                 { "IEDriverServer.exe", Resources.IEDriverServer },
                 { "MicrosoftWebDriver.exe", Resources.MicrosoftWebDriver }
             };
-            foreach (var exe in exeList)
+            foreach (var exe in driversDictionary.Where(e => !DoesFileExist(unpackfolder, e.Key) || AreFilesOfTheSameLength(e.Value.Length, unpackfolder, e.Key)))
```
Comment (lfronczyk) [inline G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: I think you forgot a exclamation mark before `AreFilesOfTheSameLength(e.Value.Length, unpackfolder, e.Key)`, so the files extracted will be the ones that didn't exist or had DIFFERENT size. I guess the same issue goes to the second PR where you use the same code.

Comment (lfronczyk) [reply G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs]: And come on, name the variables (exe & driversDictionary) better ot at least comment my remark with the reason why you won't do it. I asked you to do that twice...

**Pull request merged**